dcicutils/misc_utils.py

```diff
@@ -11,6 +13,54 @@
 PRINT = print
 
 
+class VirtualApp(TestApp):
+    """ Wrapper class for TestApp, which we use a handler to the Encoded Application where we can submit
+        requests to it simulating a number of conditions, including permissions.
+    """
+    logging.basicConfig()
+
+
+    def __init__(self, app, environ):
+        """ Builds an encoded application, allowing you to submit requests to an encoded application
+
+        :param app: return value of get_app(config_uri, app_name)
+        :param environ: options to pass to the application. Usually permissions.
+        """
+        self.virtual_app = TestApp(app, environ)
+
+    def get(self, url, **kwargs):
```

Comment:
If you inherit from `TestApp`, the way you write this is:
```
def get(self, url, **kwargs):
    """Wrapper for TestApp...etc."""
    logging.info("OUTGOING HTTP GET: %s' % url)
    super(VirtualApp, self).get(url, **kwargs)
```
The syntax may have a simpler option in Python3, but my point is you just call the super rather than calling something in a slot.